internal/kafka/test/integration/kafkas_test.go

```diff
@@ -526,20 +526,13 @@ func testValidations(h *coreTest.Helper, t *testing.T, kafkaValidations []kafkaV
 
 	for _, val := range kafkaValidations {
 		errK := test.TestServices.KafkaService.RegisterKafkaJob(val.kafka)
-		if (errK != nil) != val.eCheck.wantErr {
-			t.Errorf("Ung.Expected error %v, wantErr = %v for kafka %s", errK, val.eCheck.wantErr, val.kafka.Name)
-		}
-
+		g.Expect(errK != nil).To(gomega.Equal(val.eCheck.wantErr), fmt.Sprintf("Expected error %v, wantErr = %v for kafka %s", errK, val.eCheck.wantErr, val.kafka.Name))
 		if val.eCheck.wantErr {
 			if errK == nil {
-				t.Errorf("RegisterKafkaJob() g.Expected err to be received but got nil")
+				g.Expect(errK).To(gomega.HaveOccurred(), "RegisterKafkaJob() g.Expected err to be received but got nil")
 			} else {
-				if errK.Code != val.eCheck.code {
-					t.Errorf("RegisterKafkaJob() received error code %v, g.Expected error %v for kafka %s", errK.Code, val.eCheck.code, val.kafka.Name)
-				}
-				if errK.HttpCode != val.eCheck.httpCode {
-					t.Errorf("RegisterKafkaJob() received http code %v, g.Expected %v for kafka %s", errK.HttpCode, val.eCheck.httpCode, val.kafka.Name)
-				}
+				g.Expect(errK.Code).To(gomega.Equal(val.eCheck.code), fmt.Sprintf("RegisterKafkaJob() received error code %v, g.Expected error %v for kafka %s", errK.Code, val.eCheck.code, val.kafka.Name))
+				g.Expect(errK.HttpCode).To(gomega.Equal(val.eCheck.httpCode), fmt.Sprintf("RegisterKafkaJob() received error code %v, g.Expected error %v for kafka %s", errK.Code, val.eCheck.code, val.kafka.Name))
```

Comment:
I think we can get rid of this if/else statement here? If errK is nil, the first assertion would fail so it won't call the assertion for code and httpcode.